c10/core/GradMode.cpp

```diff
@@ -3,15 +3,11 @@
 #include <stdexcept>
 
 namespace c10 {
-
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-thread_local bool GradMode_enabled = true;
-
 bool GradMode::is_enabled() {
-  return GradMode_enabled;
+  return is_enabled(impl::_get_thread_local_state());
```

Comment:
I guess you don't need this c++ file anymore?

Reply:
We do. Several other overloads are defined in it.